### ISSUE TITLE
Coverlet is very slow

### ISSUE THREAD

**Issue by lillo42** [coverlet-coverage/coverlet#1303]
Hi,

I'm trying to add coverlet to project that I'm working now, The application is a monolith, when I run the test without code coverlet it's took around 5 min, and I run with coverlet it's taking around 20min, Is there anything that I can do run it faster?

I'm using coverlet.msbuild

The command I'm using

```bash
dotnet test -c Release --no-build --logger:"junit;LogFilePath=/var/tmp/{assembly}-test-result.xml;MethodFormat=Class;FailureBodyFormat=Verbose" -p:CollectCoverage=true -p:CoverletOutputFormat=opencover -p:Exclude="[Data.Migrations]*%2c[DbRoutines]*%2c[NServiceBus]*"
```

**Comment by petli** [comment]
Hi @lillo42, can you tell if it is instrumentation that takes time, or running the instrumented tests? 

If it is running the tests and you don't need to know how many times each line is hit, you can try adding `-p:SingleHit=true` which may reduce the overhead for you.  That option is not documented in https://github.com/coverlet-coverage/coverlet/blob/master/Documentation/MSBuildIntegration.md but is in https://github.com/coverlet-coverage/coverlet/blob/master/Documentation/VSTestIntegration.md, and I think it should work with `coverlet.msbuild` too.

**Comment by lillo42** [comment]
@petli thanks, I'll test it 

>  can you tell if it is instrumentation that takes time, or running the instrumented tests?
How can I check it?

Looking to the logs, seems to be `Calculating coverage result...`

**Comment by lillo42** [comment]
@petli the `-p:SingleHit=true` didn't work =(  but I found my issue, the problem was my migration code, when I excluded it, it's became much more faster

**Comment by petli** [comment]
@lillo42 Sounds like this was more due to the volume of code then, rather than test execution time. Good that you could solve it! I close the issue, but feel free to reopen if you need to.

**Comment by olegd-superoffice** [comment]
> I found my issue, the problem was my migration code, when I excluded it, it's became much more faster

@lillo42 Was it EF database migration code, or something else? Could you provide more details about your fix?

**Comment by lillo42** [comment]
@olegd-superoffice it was EF database migration code, unfortunately, I don't remember the code I did (I don't have access to it); I think it was:

```bash
dotnet test -c Release --no-build  /p:CollectCoverage=true /p:CoverletOutputFormat=opencover /p:Exclude="[Data.Migrations]*"
```
Look at these docs: https://github.com/coverlet-coverage/coverlet/blob/master/Documentation/MSBuildIntegration.md#excluding-from-coverage .

**Comment by olegd-superoffice** [comment]
Excluding EF Core migrations from coverage did not help me for some reason (I tried both namespace-based and file-based patterns for exclusion). However, I was able to solve the problem by squashing migrations, as described in [this blog post](https://mitchelsellers.com/blog/article/squashing-ef-core-migrations-safely). Our database has about 100 tables, and we had almost 500 EF Core migrations in our code. It was very noticeable that running tests with coverage was getting slower as we added more migrations to evolve our database. So, it definitely caused slowness, but I couldn't fix it by excluding it from coverage.
